classes/Product.php

```diff
@@ -3276,11 +3276,15 @@ public static function getPricesDrop(
         $ids_product = Product::_getProductIdByDate((!$beginning ? $current_date : $beginning), (!$ending ? $current_date : $ending), $context);
 
         $tab_id_product = [];
+
         foreach ($ids_product as $product) {
             if (is_array($product)) {
-                $tab_id_product[] = (int) $product['id_product'];
+                $productId = (int) $product['id_product'];
             } else {
-                $tab_id_product[] = (int) $product;
+                $productId = (int) $product;
+            }
```

Comment:
```suggestion
        $productId = (int) (is_array($product) ? $product['id_product'] : $product);
```

Reply:
Interesting code style improvement, but IMHO I don't think it's worth blocking the PR for this So I still approve it and leave the decision to @kusflo to integrate this modification

Reply:
Thanks for the contribution @Progi1984  but I prefer not to change it